docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,747 @@
+
+
+Records
+-------
+
+Records represent the goods being tracked by Track and Trade. Almost
+every transaction references some Record.
+
+A Record contains a unique identifier, the name of a RecordType, and
+lists containing the history of its owners and custodians. It also
+contains a ``final`` flag indicating whether further updates can be
+made to the Record and its Properties. If this flag is set to true,
+then no further updates can be made to the Record, including changing
+its ``final`` flag.
+
+.. code-block:: protobuf
+
+   message Record {
+       message AssociatedAgent {
+           // The Agent's public key
+           string agent_id = 1;
+
+           // Approximately when this agent was associated, as a Unix UTC timestamp
+           uint64 timestamp = 2;
+       }
+
+       // The unique user-defined natural key which identifies the
+       // object in the real world (for example, a serial number)
+       string identifier = 1;
+
+       string record_type = 2;
+
+       // Ordered oldest to newest by timestamp
+       repeated AssociatedAgent owners = 3;
+       repeated AssociatedAgent custodians = 4;
+
+       // Flag indicating whether the Record can be updated. If it is set
+       // to true, then the record has been finalized and no further
+       // changes can be made to it or its Properties.
+       bool final = 5;
+   }
+
+
+Note that while information about a Record's owners and custodians are
+included in the object, information about its Properties are stored
+separately (see the Properties_ section below).
+
+Records whose addresses collide are stored in a list alphabetically by
+identifier.
+
+.. code-block:: protobuf
+
+   message RecordContainer {
+       repeated Record entries = 1;
+   }
+
+.. _Properties:
+
+Properties
+----------
+
+Historical data pertaining to a particular data field of a tracked
+object are stored as Properties, represented as a list of values
+accompanied by a timestamp and a reporter identifier.
+
+The whole history of updates to Record data is stored in current state
+because this allows for more flexibility in writing transaction rules.
+For example, For example, in a fish track-and-trade system, there
+might be a rule that no fish can be exchanged whose temperature has
+gone above 40 degrees. This means, however, that it would be
+impractical to store all of a Record's data at one address, since
+adding a single update would require reading the entire history of
+each of the record's properties out of state, adding the update, then
+writing it all back.
+
+To solve this problem, Properties are stored in their own namespace
+derived from their name and associated Record. Since some Properties
+may have thousands of updates, four characters are reserved at the end
+of that namespace in order to paginate a Property's history. The
+Property itself, with name, Record identifier, authorized reporters,
+and paging information, are stored at the namespace ending in
```

Comment:
Should be "is" instead of "are".  Perhaps some judiciously placed parens would help clarify the sentence: 

     The Property itself (with name, Record identifier, authorized reporters, and paging information) is stored ...